compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala

```diff
@@ -1829,7 +1829,15 @@ class JSCodeGen()(using genCtx: Context) {
         }
 
       case Block(stats, expr) =>
-        js.Block(stats.map(genStat) :+ genStatOrExpr(expr, isStat))
+        // #15419 Collapse { <undefined-param>; BoxedUnit } to <undefined-param>
+        val genStatsAndExpr0 = stats.map(genStat(_)) :+ genStatOrExpr(expr, isStat)
```

Comment:
Minor, style: did you add the underscore on purpose?

```suggestion
        val genStatsAndExpr0 = stats.map(genStat) :+ genStatOrExpr(expr, isStat)
```

Reply:
Yes, I always add the underscore on purpose. For some reason I don't trust eta expansion :sweat_smile: